src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -117,15 +121,25 @@ kvstore::ResultCode UpdateEdgeProcessor::collectVertexProps(
             // load it. To protect the data, we just return failed to graphd.
             return kvstore::ResultCode::ERR_UNKNOWN;
         }
-        const auto constSchema = reader->getSchema();
+        const auto schema = this->schemaMan_->getTagSchema(this->spaceId_, tagId);
+        if (schema == nullptr) {
+            LOG(WARNING) << "Can't find the schema for tagId " << tagId;
+            return kvstore::ResultCode::ERR_TAG_NOT_FOUND;
+        }
         for (auto& prop : props) {
+            VariantType v;
             auto res = RowReader::getPropByName(reader.get(), prop.prop_.name);
             if (!ok(res)) {
```

Comment:
The logic is right? 
The default value order should be "default value in schema" > "default value in RowReader". right?

Reply:
Further more,  could we extract the default value logic into one place?

Reply:
You think we need to add `default value in RowReader`? For now, most logic in update/upsert only check the `default value in schema`. If failed to get it, then we return error.

Reply:
That would affect lots of space, I've discussed about it during a meeting with sherman-the-tank and jude-zhu, we just keep it for now.

One of the changes is that we need to modify all places that read from a row, which would face the problem of alter schema. In that place, we use the `default value in RowReader`. Indeed, it is inconsistent.

Reply:
Thats fine.